multinode_integration_tests/src/main.rs

```diff
@@ -342,6 +349,104 @@ impl MockNode {
     }
 }
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+struct ProbeTarget {
+    ip_address: IpAddr,
+    port_specs: Vec<PortSpec>,
+}
+
+impl FromStr for ProbeTarget {
+    type Err = String;
```

Comment:
This is actually rather a supplement to the following comment, but if I were you, I'd make the Err type a static str `&'static str`, because the always the same error messages allow it, by which you wouldn't need to write `.to_string()` after each of those (repeated) error messages.

Reply:
Okay, I'll try it. I'm handicapped by the fact that even after years of Rust, I still don't understand exactly what `'static` means. I get the feeling that almost nobody _actually_ understands it, but most people go ahead and use it anyway. Maybe I'm wrong about that.

Reply:
With `str`s, the interpretation of `'static` is easy. It de facto means the string is firmly determined for the whole life span of the program. It is hard-coded.

In other situations, especially with its concurrences in type boundaries, it translates to me as "the type cannot be referenced. It must be passed in together with the ownership. :man_shrugging: 

The hardest interpretations comes for me with trait objects, which by default has the `'static` quality if you don't specify elsehow. The fun starts with the `&dyn` and `&mut dyn` also implement it, talking about the desuggared syntax looking like `&dyn MyTrait + 'static` - that's the default. However if in your trait object, in the concrete struct hiding behind is a reference (it has a referenced field) then the trait over it cannot implement `'static` but you must introduce a specific limited lifetime say of the name `'a` like here `&dyn MyTrait + 'a`. 

Yeah, sometimes it can be a tough fight but I feel like it always has a solution.